File: client/src/app/models/media.ts
export enum MediaType {
    Image = 1,
    Video = 2,
    Model3d = 3
}

export interface CloudMedia {
    id: number,
    url: string,
    cloudId: string,
    type: MediaType
}

export interface CraftMedia extends CloudMedia {
    craftId: number,
    craftName: string
}

export interface UserMedia extends CloudMedia {
    userId: number,
    userUserName: string
}

File: API/Entities/Media.cs
using CraftyCommon.DTOs;
using System;

namespace API.Entities;


/// <summary>
/// This class represents either a photo or a video that is stored in the cloud.
/// </summary>
public class Media
{
    public long Id { get; set; }     
    public required string Url { get; set; }
    public string? CloudId { get; set; }
    public required MediaType Type { get; set; } = MediaType.Image;
}


public class CraftMedia : Media
{
    public long CraftId { get; set; }
    public Craft? Craft { get; set; }
}

public class UserMedia : Media
{
    public long UserId { get; set; }
    public User? User { get; set; }
}